### ISSUE TITLE
no stack trace for some errors outside `@test`

### ISSUE THREAD

**Issue by JeffBezanson** [JuliaLang/julia#26155]
E.g.

```
Error in testset FileWatching:
Error During Test at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\test\testdefs.jl:19
  Got an exception of type LoadError outside of a @test
  LoadError: FDWatcher: socket operation on non-socket (ENOTSOCK)
  in expression starting at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\site\v0.7\FileWatching\test\runtests.jl:77
ERROR: LoadError: Test run finished with errors
```

This makes it basically impossible to diagnose such failures.

**Comment by JeffBezanson** [comment]
OK, this does work in many cases, just not for that particular error. The error might come from a callback, which could be related.

**Comment by samoconnor** [comment]
Is it related to this: https://github.com/JuliaLang/julia/issues/19979#issuecomment-344439163 ?

**Comment by c42f** [comment]
Possibly fixed by #28878 ?

**Comment by vtjnash** [comment]
Or possibly fixed by https://github.com/JuliaLang/julia/pull/44995, since I think `@test`/`@testset` previously used to strip out all backtrace info from the user
